frame/support/src/traits/hooks.rs

```diff
@@ -227,10 +227,16 @@ pub trait Hooks<BlockNumber> {
 	fn on_finalize(_n: BlockNumber) {}
 
 	/// This will be run when the block is being finalized (before `on_finalize`).
-	/// Implement to have something happen using the remaining weight.
-	/// Will not fire if the remaining weight is 0.
-	/// Return the weight used, the hook will subtract it from current weight used
-	/// and pass the result to the next `on_idle` hook if it exists.
+	///
+	/// Implement to have something happen using the remaining weight. Will not fire if the
+	/// remaining weight is 0.
+	///
+	/// The order of the pallets in the runtime enum does not affect the order in which each
+	/// pallet's `on_idle` hook is called. Instead, `on_idle` hooks are called in a round-robin
+	/// fashion indexed by the block number.
```

Comment:
Technically, the order of the pallets in the runtime does affect which pallet's on_idle is called. 😁 
```suggestion
	/// The order of the pallets in the runtime enum determines the order in which each pallet's
	/// `on_idle` hook is called with each pallet `on_idle` hook getting a chance to be chosen the
	/// first to execute in a round robin fashion indexed by the block number.
```

Reply:
@Ank4n do you mean that in normal circumstances the order of the pallets in the runtime affects the order of the pallet being called? I'm trying to say that the behaviour for the `on_idle` hook is different from the default.

Reply:
But yes, technically, the ordering still matter though, but the first `on_idle` to be called is selected using the round robin. I agree this is more accurate.

Reply:
I think this may be a more accurate and concise writeup.
```
/// Each pallet's `on_idle` is chosen to be the first to execute in a round-robin fashion
/// indexed by the block number. While weight remains, next hooks in a cyclic order of the pallets
/// in the `Runtime` enum are executed.
```